### PR TITLE
Added more validation tests

### PR DESCRIPTION
**Pull request by wonga00**
renamed issueManager.js to issue_manager.js for consistency
added tests for IssueManager,  and deprecatedTag, disconnectedHighway, tagSuggestsArea validators

### CODE REVIEW

File: test/spec/validations/deprecated_tag.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        var issues = validate();
+        expect(issues).to.have.lengthOf(0);
+    });
+
+    it('finds deprecated tags', function() {
+        createWay({'highway': 'ford'});
+        var issues = validate();
+        expect(issues).to.have.lengthOf(1);
+        var issue = issues[0];
+        expect(issue.type).to.eql(iD.ValidationIssueType.deprecated_tags);
```
Comment (gaoxm) [inline test/spec/validations/deprecated_tag.js]: add a check on severity?

File: test/spec/validations/disconnected_highway.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        var issues = validate();
+        expect(issues).to.have.lengthOf(0);
+    });
+
+    it('finds disconnected highway', function() {
+        createWay({'highway': 'unclassified'});
+        var issues = validate();
+        expect(issues).to.have.lengthOf(1);
+        var issue = issues[0];
+        expect(issue.type).to.eql(iD.ValidationIssueType.disconnected_highway);
```
Comment (gaoxm) [inline test/spec/validations/disconnected_highway.js]: ditto

File: test/spec/validations/issue_manager.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        createInvalidWay();
+        var issueManager = new iD.IssueManager(context);
+        var issues = issueManager.getIssues();
+        expect(issues).to.have.lengthOf(0);
+
+        issueManager.validate();
+        issues = issueManager.getIssues();
+        expect(issues).to.have.lengthOf(1);
+        var issue = issues[0];
+        expect(issue.type).to.eql(iD.ValidationIssueType.missing_tag);
```
Comment (gaoxm) [inline test/spec/validations/issue_manager.js]: ditto

File: test/spec/validations/tag_suggests_area.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        var issues = validate();
+        expect(issues).to.have.lengthOf(0);
+    });
+
+    it('finds tags that suggest area', function() {
+        createWay({'building': 'yes'});
+        var issues = validate();
+        expect(issues).to.have.lengthOf(1);
+        var issue = issues[0];
+        expect(issue.type).to.eql(iD.ValidationIssueType.tag_suggests_area);
```
Comment (gaoxm) [inline test/spec/validations/tag_suggests_area.js]: ditto

File: test/spec/validations/issue_manager.js (hunk 3 above)
Comment (wonga00) [reply test/spec/validations/issue_manager.js]: i think this one is OK -- i'm just testing that the issuemanager validates

**Pull request merged**